src/util/customTransformer-faas.js

```diff
@@ -1,19 +1,72 @@
 const { v4: uuidv4 } = require('uuid');
+const crypto = require('crypto');
+const NodeCache = require('node-cache');
 const stats = require('./stats');
 const { getMetadata } = require('../v0/util');
-const { setupFaasFunction, executeFaasFunction } = require('./openfaas');
+const { setupFaasFunction, executeFaasFunction, FAAS_AST_FN_NAME, FAAS_AST_VID } = require('./openfaas');
+const { getLibraryCodeV1 } = require('./customTransforrmationsStore-v1');
+
+const libVersionIdsCache = new NodeCache();
 
 function generateFunctionName(userTransformation, testMode) {
+  if (userTransformation.versionId === FAAS_AST_VID) return FAAS_AST_FN_NAME;
+
   if (testMode) {
-    const funcName = `fn-test-${userTransformation.testName?.replace('_', '-')}-${uuidv4()}`;
+    const funcName = `fn-test-${uuidv4()}`;
     return funcName.substring(0, 63).toLowerCase();
   }
 
-  return `fn-${userTransformation.workspaceId}-${userTransformation.versionId}`
+  userTransformation.libraryVersionIds = userTransformation.libraryVersionIds || [];
+  const ids = [userTransformation.workspaceId, userTransformation.versionId].concat(userTransformation.libraryVersionIds.sort());
+  const hash = crypto.createHash('md5').update(`${ids}`).digest('hex');
+
+  return `fn-${userTransformation.workspaceId}-${hash}`
     .substring(0, 63)
     .toLowerCase();
 }
 
+async function extractRelevantLibraryVersionIdsForVersionId(functionName, code, versionId, libraryVersionIds, prepopulatedImports) {
+  if (functionName === FAAS_AST_FN_NAME || versionId == FAAS_AST_VID) return [];
+
+  const cachedLvids = libVersionIdsCache.get(functionName);
+
+  if (cachedLvids) return cachedLvids;
+
+  const libraries = await Promise.all(
+    (libraryVersionIds || []).map(async (libraryVersionId) => getLibraryCodeV1(libraryVersionId)),
+  );
+
+  const relevantLvids = [];
+
+  if (libraries) {
+    let extractedLibraries;
+
+    if (Array.isArray(prepopulatedImports)) {
+      extractedLibraries = prepopulatedImports
+    } else {
+      extractedLibraries = Object.keys(await require('./customTransformer').extractLibraries(
```

Comment:
This step looks redundant. To avoid this, we have done data migration step already.
Also it doesn't propagate to else condition, given default value for imports is empty list